File: app/src/main/java/com/keelim/nandadiagnosis/db/NandaDao.java
package com.keelim.nandadiagnosis.db;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
interface NandaDao {

    @Query("SELECT * FROM nanda")
    List<Nanda> getAllNanda();

    @Query("DELETE FROM nanda ")
    void clearAll();

    @Insert
    void insert(Nanda nanda);

    @Update
    void update(Nanda nanda);

    @Delete
    void delete(Nanda nanda);
}

File: shared/src/commonMain/kotlin/com/keelim/shared/data/database/dao/NandaDao.kt
package com.keelim.shared.data.database.dao

import androidx.room.Dao
import androidx.room.Query
import com.keelim.shared.data.database.model.NandaEntity
import kotlinx.coroutines.flow.Flow

@Dao
interface NandaDao {
    @Query("SELECT * FROM nanda")
    fun getNandaEntities(): Flow<List<NandaEntity>>

    @Query("SELECT * FROM nanda WHERE reason LIKE '%' || :query || '%'")
    fun getDiagnosis(query: String): Flow<List<NandaEntity>>
}
